curriculum/challenges/english/25-front-end-development/workshop-recipe-tracker/66fbcf750a62784cf11f5630.md

```diff
@@ -13,30 +13,56 @@ You should now push the three objects into the `recipes` array. To do thus, you
 
 Use the `push()` method to push all the recipe objects into the `recipes` array. Make sure to push `recipe1`, `recipe2`, and `recipe3` in that order.
 
+Also delete `recipe1Name`, `recipe2Name`, `recipe1CookingTime`, and `recipe2CookingTime` variables and their logging to console.
+
 # --hints--
 
-You should remove `recipe1Name` and its console log.
+You should remove `recipe1Name` variable.
+
+```js
+assert.notMatch(__helpers.removeJSComments(code), /const\s+recipe1Name\s*=\s*recipe1\s*\.\s*name\s*;?/);
+```
+
+You should remove `console.log(recipe1Name);`.
+
+```js
+assert.notMatch(__helpers.removeJSComments(code), /console\s*\.\s*log\s*\(\s*recipe1Name\s*\)\s*;?/);
+```
+
+You should remove `recipe2Name` variable.
+
+```js
+assert.notMatch(__helpers.removeJSComments(code), /const\s+recipe2Name\s*=\s*recipe2\s*\.\s*name\s*;?/);
```

Comment:
```suggestion
assert.isUndefined(recipe2name);
```

I think these will work, but it's possible they will throw a reference error. I can't remember.

Reply:
Unfortunately it will cause `ReferenceError`. We could do it with `try/catch` instead, ie:

```js
try {
  recipe1Name;
  assert.fail('Variable is defined/declared');
} catch (e) {
  assert.include(e?.message, 'recipe1Name is not defined');
}
```

What do you think?

Reply:
That works!